CMakeLists.txt

```
@@ -0,0 +1,188 @@
+cmake_minimum_required(VERSION 3.9 FATAL_ERROR)
+project(rubberband VERSION 2.1.1)
+
+option(RUBBERBAND_EXAMPLE "Build the Rubberband example app" OFF)
+option(RUBBERBAND_OPTIMIZED "Build Rubberband optimized for the build machine's CPU" OFF)
+### Dependencies ###
+
+# ARM cross-compilers do not support march=native, only mtune=native
+set(MARCH_NATIVE_FLAG)
+
+include(CheckCXXCompilerFlag)
+
+if(RUBBERBAND_OPTIMIZED)
+  check_cxx_compiler_flag("-march=native" MARCH_NATIVE_SUPPORTED)
+  if(MARCH_NATIVE_SUPPORTED)
+    set(MARCH_NATIVE_FLAG "-march=native")
+  else()
+  endif()
+endif()
+
+find_package(Threads REQUIRED)
+
+# FFTW3
+find_path(FFTW3_INCLUDEDIR fftw3.h)
+find_library(FFTW3_LIBRARY fftw3)
+
+# libsamplerate
+if(NOT TARGET samplerate)
+  find_path(SAMPLERATE_INCLUDEDIR samplerate.h)
+  find_library(SAMPLERATE_LIBRARY samplerate)
+endif()
+
+# Accelerate framework
+if(APPLE)
+  find_library(ACCELERATE_FRAMEWORK NAMES Accelerate)
+endif()
+
+### Create our target ###
+add_library(rubberband)
+
+### Handle general flags, include paths, etc ###
+target_compile_options(rubberband
+  PRIVATE
+    $<$<BOOL:${APPLE}>:-mmacosx-version-min=10.11>
+    $<$<AND:$<NOT:$<CXX_COMPILER_ID:MSVC>>,$<CONFIG:Release>>:-ffast-math ${MARCH_NATIVE_FLAG} -O3 -ftree-vectorize>
```

I personally find these generator expressiong very hard to read. This would be much more straightforward IMHO:

```suggestion
if(APPLE)
  target_compile_options(rubberband PRIVATE -mmacosx-version-min=10.11)
elif(NOT MSVC)
    target_compile_options(rubberband PRIVATE $<CONFIG:Release:-ffast-math ${MARCH_NATIVE_FLAG} -O3 -ftree-vectorize>)
```

I'll open a PR.